wasm/src/lib.rs

```diff
@@ -37,6 +38,48 @@ pub fn git_hash() -> String {
     git_hash.into()
 }
 
+fn get_qir_internal(code: &str) -> std::result::Result<String, (Error, Vec<Frame>)> {
```

Comment:
I was starting to make a suggestion that involved following the pattern of `run_internal` and using `stateless::Error` to return either an eval failure or compilation failure (since it has an `ErrorKind` for each of those), but changing the return type and marshalling the errors is too big of a change for a git-diff suggestion. I think we should circle back to that for a follow-up PR that cleans up the pattern here; yes, as the comment points out below, the only caller we have currently verifies the document state before calling into this, but since we have to call compile we might as well use the error kinds we have to return at least the first compilation error like `run_internal`.